pytests/operators/test_stateful.py

```diff
@@ -284,8 +284,57 @@ def test_stateful_snapshots_discard_per_key(recovery_config):
     ]
 
     out.clear()
-    run_main(flow, epoch_interval=ZERO_TD, recovery_config=recovery_config)
+    run_main(flow, epoch_interval=ZERO_TD, recovery_config=recovery_config_immediate)
     assert out == [
         ("a", (None, "a3")),
         ("b", ("b2", "b3")),
     ]
+
+
+def test_stateful_recovers_older_snapshots(recovery_config_immediate):
```

Comment:
This was added as a regression test for a bug I had while implementing the fixes, where the local state store was not correctly initialized and only showed on subsequent resumes. It's a bug that was not there before, and should not be here now, but I thought of leaving this here to ensure it doesn't show up in future refactorings, since other tests where not checking this specific case.